control/controlclient/debug.go

```diff
@@ -42,28 +43,51 @@ func dumpGoroutinesToURL(c *http.Client, targetURL string) {
 	}
 }
 
-var reHexArgs = regexp.MustCompile(`\b0x[0-9a-f]+\b`)
+func isHexString(s string) bool {
+	if len(s) <= 2 || s[0:2] != "0x" {
+		return false
+	}
+	_, err := strconv.ParseUint(s[2:], 16, 64)
+	return err == nil
+}
+
+func separator(r rune) bool {
+	if unicode.IsLetter(r) || unicode.IsNumber(r) {
+		return false
+	}
+	return true
+}
+
+func scrubGoroutineDump(buf []byte) []byte {
+	saw := map[string]string{} // "0x123" => "v1%3" (unique value 1 and its value mod 8)
+	sbuf := string(buf)
```

Comment:
I'd like to avoid this allocation too. The input []byte is already quite big and this doubles it.